### PR TITLE
Add optional umami configurations

### PR DESCRIPTION
**Pull request by EloEntropy**
Minor pull request to include an optional "doNotTrack" config option to respect client Do Not Track requests.

### CODE REVIEW

**Comment by spekulatius** [comment]
Hello @EloEntropy 

I found the configuration in the docs (https://umami.is/docs/tracker-config) but I'm not entirely sure what the tracker would do in this case. Do you know more about this?

Cheers,
Peter

**Comment by EloEntropy** [comment]
Hi Peter,

If a visitor sets the Do Not Track option in their local browser, then the umami tracker will check this before choosing to send data back to the server. So if you enable it in the tracker config, it will respect the visitor's privacy request. 

I saw the configuration in the docs as well, and wanted to implement it for my own page. I just thought it might be worth pushing back to your repo in case other people might also want to have the option.

An example of a Firefox Do Not Track setting is below, but you can find this option in most browsers.

image

Hope that answers your question!

File: inject.js [hunk 1]
```diff
@@ -14,8 +14,14 @@ export default ({ router }) => {
       var d=document,
       g=d.createElement('script'),
       s=d.getElementsByTagName('script')[0];
-
+
```
Comment (spekulatius) [inline inject.js]: Can you remove the spaces here @EloEntropy ?

**Review by spekulatius — CHANGES_REQUESTED** [review]

**Comment by spekulatius** [comment]
Do you think it makes sense to implement the [other options](https://umami.is/docs/tracker-config) as well @EloEntropy ?

File: inject.js (hunk 1 above)
Comment (EloEntropy) [reply inject.js]: I have removed the line in the latest commit.

**Comment by EloEntropy** [comment]
I considered it before, but I agree that it would be an improvement to add all of the options. At least to allow the plugin user to make the choice if they need it. I have included my crude attempt at adding the rest in the latest commit, but if you have a cleaner method feel free to let me know. I appreciate the original plugin creation 👍  @spekulatius

**Comment by spekulatius** [comment]
Hello @EloEntropy 

looks good! Thank you for adding the other options!

Cheers,
Peter

**Pull request merged**